.github/workflows/pr-title.yml

```diff
@@ -11,10 +11,22 @@ jobs:
   main:
     name: Validate PR title
     runs-on: ubuntu-latest
+    permissions:
+      actions: none
+      checks: none
+      contents: none
+      deployments: none
+      id-token: none
+      issues: none
+      packages: none
+      pages: none
+      pull-requests: none
+      repository-projects: none
+      security-events: none
+      statuses: none
     steps:
-      # Please look up the latest version from
-      # https://github.com/amannn/action-semantic-pull-request/releases
-      - uses: amannn/action-semantic-pull-request@v3.6.0
+      # Tag v4.2.0 https://github.com/amannn/action-semantic-pull-request/commit/0eb081bc9c35210408951834a444794406eff6f8
+      - uses: amannn/action-semantic-pull-request@ad47c64a77f00b53fe0a6b41d9a65f14c2e0b5fd
```

Comment:
Any explanation why using the commit instead of the tag itself?

Reply:
Should the `amannn/action-semantic-pull-request` repository be compromised by an attacker, they would be able to push new code and change the target commit of any tag.

By referencing the commit SHA, we make this theoretical attack more complex: the attacker would need to forge a commit with the same hash.